vdr/didx509/resolver.go

```diff
@@ -115,12 +116,29 @@ func (r Resolver) Resolve(id did.DID, metadata *resolver.ResolveMetadata) (*did.
 		return nil, nil, err
 	}
 
+	// Validate certificate chain, checking signatures and whether the chain is complete
+	var chainWithoutLeaf []*x509.Certificate
+	for _, curr := range chain {
+		if curr.Equal(validationCert) {
+			continue
+		}
+		chainWithoutLeaf = append(chainWithoutLeaf, curr)
+	}
+	trustStore := core.BuildTrustStore(chainWithoutLeaf)
+	verifiedChains, err := validationCert.Verify(x509.VerifyOptions{
+		Intermediates: core.NewCertPool(trustStore.IntermediateCAs),
+		Roots:         core.NewCertPool(trustStore.RootCAs),
+	})
+	if err != nil {
+		return nil, nil, fmt.Errorf("did:509 certificate chain validation failed: %w", err)
+	}
+
 	err = validatePolicy(ref, validationCert)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	err = r.pkiValidator.ValidateStrict(chain)
+	err = r.pkiValidator.ValidateStrict(verifiedChains[0])
```

Comment:
this only works if the leaf cert itself is revoked. I think the entire chain should be checked to make sure intermediates are not revoked?

Reply:
nah, validateCert verifies the whole chain. But `verifiedChains` is defined as `verifiedChains [][]x509.Certificate` (it's a slice of chains)